### PR TITLE
IPv4/single: SAME70 EMAC buffer sizes

### PR DESCRIPTION
**Pull request by htibosch**
Description
-----------

In the [DriverSAM](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/main/source/portable/NetworkInterface/DriverSAM) driver, the macro `GMAC_FRAME_LENTGH_MAX` is the maximum supported frame length, 1536 bytes. That is the maximum number of bytes in an Ethernet packet.

The macro `NETWORK_BUFFER_SIZE` was supposed to define size of a network buffer, which should be 1536 bytes **plus** a `ipBUFFER_PADDING` (10) hidden bytes .

This PR makes sure that the statically allocated buffers in `ucNetworkPackets[]` are large enough to accommodate the largest possible incoming packet.

To summarize this PR:
~~~diff
-#define NETWORK_BUFFER_SIZE    ( 1536 )
+#define NETWORK_BUFFER_SIZE    ( 1536+16 )
~~~

`ipBUFFER_PADDING` is rounded up to 16 to get a nice alignment.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by htibosch** [comment]
In "ICMP checksum calculated manually" I re-enabled the function `vGMACGenerateChecksum()`, but only for ICMP protocol checksum.
As for the SAME70, it does not have to calculate UDP or TCP protocol checksums.

**Review by AniruddhaKanhere — DISMISSED** [review]

**Review by shubnil — DISMISSED** [review]

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by shubnil — APPROVED** [review]
